### PR TITLE
Clean up Uninstall output

### PR DESCRIPTION
**Pull request by rcritten**
ipa-server-install did not configure a console logger. This caused odd output because ipa-client-install does and a normal server uninstall would end with "ipa-client-install command was successful"

Because there was no console logger there were print statements in a lot of places. Those were dropped because the equivalent logging statement was already there.

A number of info loggers were converted to debug to suppress their output. They were never shown in the past for the most part with the exception of the client installer and Forwarding and trying statements.

Adding the logger exposed two additional issues:

1. When deleting the last server the server-del command threw an LDAP error trying to write an empty attribute
2. Two ACI parsing issues related to subtypes

### CODE REVIEW

File: ipaserver/plugins/server.py [hunk 1]
```diff
@@ -619,7 +619,10 @@ def _remove_server_principal_references(self, master):
             if master in srvlist:
                 srvlist.remove(master)
                 attr = ' '.join(srvlist)
-                ret['defaultServerList'] = attr
+                if len(srvlist) == 0:
```
Comment (stlaz) [inline ipaserver/plugins/server.py]:
```python
if not srvlist:
    del ret['defaultServerList']
else:
    ret['defaultServerList'] = ' '.join(srvlist)
```
^-- removes the possibly unnecessary `join()` if the list is empty and the condition is a bit more Pythonic I believe :)

Comment (rcritten) [reply ipaserver/plugins/server.py]: +1, done

File: ipapython/install/cli.py [hunk 2]
```diff
@@ -58,7 +58,24 @@ def _get_usage(configurable_class):
 
 def install_tool(configurable_class, command_name, log_file_name,
                  debug_option=False, verbose=False, console_format=None,
-                 use_private_ccache=True, uninstall_log_file_name=None):
+                 use_private_ccache=True, uninstall_log_file_name=None,
+                 ignore_return_codes=[]):
```
Comment (tiran) [inline ipapython/install/cli.py]: Please don't use mutable data types in argument lists. If you need an iterable, use a tuple as default argument ``ignore_return_codes=()``.

File: ipalib/aci.py [hunk 3]
```diff
@@ -25,7 +25,8 @@
 # The Python re module doesn't do nested parenthesis
 
 # Break the ACI into 3 pieces: target, name, permissions/bind_rules
-ACIPat = re.compile(r'\(version\s+3.0\s*;\s*ac[li]\s+\"([^\"]*)\"\s*;\s*([^;]*);\s*\)', re.UNICODE)
+ACIPat = re.compile(r'\(version\s+3.0\s*;\s*ac[li]\s+\"([^\"]*)\"\s*;'
```
Comment (tiran) [inline ipalib/aci.py]: This change looks unrelated to the rest.

File: ipapython/admintool.py [hunk 4]
```diff
@@ -93,6 +93,7 @@ class AdminTool(object):
     log_file_name = None
     usage = None
     description = None
+    ignore_return_codes = []
```
Comment (tiran) [inline ipapython/admintool.py]: Same here, use a tuple ``()``.

File: ipatests/test_ipalib/test_aci.py [hunk 5]
```diff
@@ -162,3 +162,15 @@ def test_aci_parsing_8():
 def test_aci_parsing_9():
     check_aci_parsing('(targetfilter = "(|(objectClass=person)(objectClass=krbPrincipalAux)(objectClass=posixAccount)(objectClass=groupOfNames)(objectClass=posixGroup))")(targetattr != "aci || userPassword || krbPrincipalKey || sambaLMPassword || sambaNTPassword || passwordHistory")(version 3.0; acl "Account Admins can manage Users and Groups"; allow (add, delete, read, write) groupdn = "ldap:///cn=admins,cn=groups,cn=accounts,dc=greyoak,dc=com";)',
         '(targetattr != "aci || userPassword || krbPrincipalKey || sambaLMPassword || sambaNTPassword || passwordHistory")(targetfilter = "(|(objectClass=person)(objectClass=krbPrincipalAux)(objectClass=posixAccount)(objectClass=groupOfNames)(objectClass=posixGroup))")(version 3.0;acl "Account Admins can manage Users and Groups";allow (add,delete,read,write) groupdn = "ldap:///cn=admins,cn=groups,cn=accounts,dc=greyoak,dc=com";)')
+
+
+def test_aci_parsing_10():
```
Comment (tiran) [inline ipatests/test_ipalib/test_aci.py]: This also looks unrelated.

File: ipaserver/install/cainstance.py [hunk 6]
```diff
@@ -1794,7 +1794,7 @@ def _create_dogtag_profile(profile_id, profile_data, overwrite):
         # import the profile
         try:
             profile_api.create_profile(profile_data)
-            logger.info("Profile '%s' successfully migrated to LDAP",
+            logger.debug("Profile '%s' successfully migrated to LDAP",
```
Comment (tiran) [inline ipaserver/install/cainstance.py]: The second like with ``profile_id`` needs to be reindented, too.

**Comment by tiran** [comment]
Thanks for cleanup up the logger! :+1:

File: ipalib/aci.py (hunk 3 above)
Comment (rcritten) [reply ipalib/aci.py]: This and the change to test_aci.py are related because by enabling logging two ACI parsing errors were exposed by the LDAP updater.

ACI parsing is forgiving and errors are not treated as fatal so this didn't cause any operational issues but two rather nasty looking errors would appear while applying updates as a direct result of my output changes so I fixed this as well.

**Comment by tiran** [comment]
```
************* Module ipaserver.install.custodiainstance
ipaserver/install/custodiainstance.py:87: [W0612(unused-variable), get_custodia_instance] Unused variable 'ldap_uri')
```

**Comment by Tiboris** [comment]
master:

* b96906156be37a7b29ee74423b82f04070c84e22 Improve console logging for ipa-server-install
* 8ea227451f4d85dbca6a331a607d25744e85121b Drop attr defaultServerList if removing the last server
* 00ddb5dd53a2eb21cd7c97167b8a5f87ab728b07 server install: drop some print statements, change log level
* 036d51d5143a142e6a3070e6328a7bcd9b2125f0 Handle subyptes in ACIs
